src/components/tasks/card/index.tsx

```diff
@@ -122,7 +133,46 @@ const Card: FC<Props> = ({
       </span>
       )
   }
- 
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { requestPromise } = fetch({ url: SELF_URL });
+        const { data } = await requestPromise;
+        const { admin:adminUser, super_user: superUser } = data?.roles
+        setIsUserAuthorized(!!adminUser || !!superUser);
+      } catch (err: any) {
+        toast(ERROR, err.message);
+      }
+    };
+    fetchData();
+
+    return (() => {
+      setIsUserAuthorized(false);
+    });
+  }, []);
+
+  function handleKeyDown(event : KeyboardEvent): void {
+    if (event.altKey) {
+      setShowEditButton(true);
+    }
+  }
+
+  const debouncedHandler = useDebounce(handleKeyDown, 300);
+
+  useEffect(() => {
+
+    document?.addEventListener('keydown', debouncedHandler);
+
+    return () => {
+      document?.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
+
```

Comment:
This should not be handled by `Card` component. It should be reusable enough to be used in any component by DI.